yarn-project/circuits.js/src/structs/recursive_proof.ts

```diff
@@ -19,7 +19,7 @@ export class RecursiveProof<N extends number> {
     /**
      * Holds the serialized proof data in an array of fields, this is without the public inputs
      */
-    public proof: Tuple<Fr, N>,
+    public proof: Fr[],
```

Comment:
Changing it to an array because typescript will break by huge tuples. It won't transpile if we try to create the proof for avm, which has > 3000 fields. To be specific, calling `makeTuple(3000, fr)` will result in this error:
`Type instantiation is excessively deep and possibly infinite`